WeaselTSF/Composition.cpp

```diff
@@ -294,14 +309,10 @@ void WeaselTSF::_UpdateComposition(ITfContext *pContext)
 			}
 			_StartComposition(pContext, _fCUASWorkaroundEnabled && !config.inline_preedit);
 		}
-		else if (!status.composing && _IsComposing())
+		else if (!status.composing && composing)
 			_EndComposition(pContext);
-		if (_IsComposing() && config.inline_preedit)
+		if (composing && config.inline_preedit)
```

Comment:
During this function lifetime `_IsComposing()` may also be outdated because `EditSession` could be evaluated async and `WeaselTSF::_pComposition` won't change immediately after `_StartComposition()` or `_EndComposition()` called.

In fact, use `_IsComposing()` here can cause crash in notebook program since because of multiple release to same composition. So `composing` is a monkey patch to sync the status immediately .

Reply:
Oh, really? It uses the `TF_ES_SYNC` option when `RequestEditSession()`.
https://github.com/rime/weasel/pull/109/files/7275176b0247f382bb15c4609801c6609207bf4d#diff-585a42c603c2ce74ea23c8bbc087bc0bR76
That made me think it synchronously resulted in the change of `_pComposition`.

Reply:
That's what also confused me...

Reply:
Indeed, I found that `_StartComposition()` and `_EndComposition()` do change `_pComposition` for sync.

What resulted in crashes is `CInlinePreeditEditSession::DoEditSession` due to a illegal memory access happened in `msctf.dll` , when both commit and preedit exist. This only happened in `notepad.exe`.

I don't know what magic m$ have done there, when I tried a debugger step into the code, it do not crash. The only thing I know is that the workaround in this PR does work.